### PR TITLE
Add remove, range, and rename function

### PR DESCRIPTION
**Pull request by TonyLianLong**
I added the `remove` function (https://github.com/queryverse/NamedTupleUtilities.jl/issues/1). Tests needed to be written.

### CODE REVIEW

**Comment by TonyLianLong** [comment]
I also added the `range` function with the tests.

**Comment by TonyLianLong** [comment]
Added `rename` function with tests(https://github.com/queryverse/NamedTupleUtilities.jl/issues/4).

File: src/NamedTupleUtilities.jl [hunk 1]
```diff
@@ -1,3 +1,87 @@
...
+    typesArray = Any[]
+    vals = Any[]
+    for n in an
+        if n == bn
+            names = (names..., cn)
+        else
+            names = (names..., n)
+        end
+        typesArray = Any[typesArray..., fieldtype(a, n)]
+        vals = Any[vals..., :(getfield(a, $(QuoteNode(n))))]
```
Comment (davidanthoff) [inline src/NamedTupleUtilities.jl]: Generally, to add an element to a vector, use ``push!(array, val)``, so here for example ``push!(vals, :(getfield(a, $(QuoteNode(n)))))``.

File: src/NamedTupleUtilities.jl [hunk 2]
```diff
@@ -1,3 +1,87 @@
...
+
+julia> rename((a = 1, b = 2, c = 3),Val(:m),Val(:d))
+(a = 1, b = 2, c = 3)
+
+julia> rename((a = 1, b = 2, c = 3),Val(:a),Val(:c))
+ERROR: duplicate field name in NamedTuple: "c" is not unique
+```
+"""
+@generated function rename(a::NamedTuple{an}, ::Val{bn}, ::Val{cn}) where {an, bn, cn}
+    names = ()
```
Comment (davidanthoff) [inline src/NamedTupleUtilities.jl]: Use a vector here.

File: src/NamedTupleUtilities.jl [hunk 3]
```diff
@@ -1,3 +1,87 @@
...
+    typesArray = Any[]
+    vals = Any[]
```
Comment (davidanthoff) [inline src/NamedTupleUtilities.jl]: Type these vectors.

File: src/NamedTupleUtilities.jl [hunk 4]
```diff
@@ -1,3 +1,87 @@
...
+        if n == bn
+            names = (names..., cn)
+        else
+            names = (names..., n)
+        end
+        typesArray = Any[typesArray..., fieldtype(a, n)]
+        vals = Any[vals..., :(getfield(a, $(QuoteNode(n))))]
+    end
+    types = Tuple{typesArray...}
+    :( NamedTuple{$names,$types}(($(vals...),)) )
```
Comment (davidanthoff) [inline src/NamedTupleUtilities.jl]: Use ``return``.

**Review by davidanthoff — CHANGES_REQUESTED** [review]
Excellent, this is really great!

I've added a lot of nitpicking comments. I think we could merge your original version, as it strikes me as correct. But I think it might make sense that I generally give this kind of detailed feedback, it might help you guys get more familiar with general julia programming tricks!

**Comment by TonyLianLong** [comment]
I changed the style of the code and also added `startswith`, `endswith`, and `occursin` to it.

**Review by davidanthoff — APPROVED** [review]

**Comment by davidanthoff** [comment]
Hurray!

**Pull request merged**